src/TreeView/index.tsx

```diff
@@ -826,41 +1105,41 @@ const handleKeyDown = ({
       case "ArrowUp": {
         event.preventDefault();
         const previous = getPreviousAccessible(data, id, expandedIds);
-        if (previous != null || !disabledIds.has(previous)) {
+        if (previous != null && !disabledIds.has(previous)) {
```

Comment:
@dgreene1 Can you validate this change and the one on 1129 is good please?

Reply:
```suggestion
        if (previous != null || (previous == null && !disabledIds.has(previous))) {
```
I'd have to do testing, but this gross suggestion I'm making is the most literal translation of the JS version. Like this should be have the same way that it did before.

The issue with your first pass (`previous != null && !disabledIds.has(previous)`) is that it doesn't allow nulls in where `previous` is `null` and `disabledIds` has null. But since you changed the type to enforce that `disabledIds` is now `Set<number>` (when it was previously `Set<number | null>` then I think both your work here or my suggestion have the same ultimate result.

tl;dr: you're good to go either way.